environments/staging.yml

```diff
@@ -16,8 +16,65 @@ ingress:
   hosts:
     - snooty-data-api.docs.staging.corp.mongodb.com
 
+# Current limits and requests are based around metrics during expected peaks
 resources:
   limits:
-    memory: 1300Mi
+    memory: 3Gi
+    cpu: 2
   requests:
     memory: 1100Mi
+    cpu: 400m
+
+# The current probes are pretty lenient to avoid prematurely shutting down
+# pods during mass fetches. They should be adjusted accordingly in the future
+probes:
+  enabled: true
+  path: /liveness
+  headers: {}
+  # Determines if the pod is still alive. The pod will shut down if this fails.
+  liveness:
+    httpGet: true
+    initialDelaySeconds: 10
+    periodSeconds: 30
+    # Ideally, timeout would be lower, but this prevents any latency during memory spikes
+    timeoutSeconds: 10
+    successThreshold: 1
+    # Keeping failureThreshold for liveness higher than readiness to ensure
+    # pod stays up even if it's not ready
+    failureThreshold: 6
+  # Determines when the pod is ready to receive traffic. This may help with ensuring that requests
+  # don't disappear into the void when routed to a new pod that isn't ready yet
+  readiness:
+    httpGet: true
+    initialDelaySeconds: 10
+    periodSeconds: 30
+    timeoutSeconds: 10
+    successThreshold: 1
+    failureThreshold: 3
+
+autoscaling:
+  apiVersion: autoscaling/v2
+  minReplicas: 2
+  maxReplicas: 5
+  metrics:
+    - type: Resource
+      resource:
+        # Use cpu instead of memory since it's a more accurate metric for current usage for our API
+        name: cpu
```

Comment:
not super familiar with HPA policies but is it possible to also scale up/scale down on memory limits utilization? (since this issue has originated from memory issues)

Reply:
Yes, it is possible. I originally tried it with just memory, but the drawback was that there were rare instances in which the pods wouldn't scale down at all due to how memory was allocated to certain pods. Part of why I went with CPU was because it seemed like a more accurate indicator of current API usage/load vs. memory, which can remain untouched by the garbage collector even when there are no requests.

I think I'll add it back but keep the threshold high as a fallback? Will ping when implemented and tested!

Reply:
Tried it out with both and things seem to look good. It stabilizes at around 2-3 pods, which I think is reasonable for our use case. If interested, can take a look at [Grafana](https://grafana.staging.corp.mongodb.com/d/a164a7f0339f99e89cea5ab47f9be872/memory-profiler-workloads?var-datasource=Prometheus&var-cluster=&var-namespace=docs&var-workload=snooty-data-api-web-app&var-type=Deployment&from=1700667742808&to=1700678860306&orgId=1). I tried to be a little lenient with memory utilization, but I think we can always tweak as we observe its usage in prod 👍

Reply:
looks good to me!